terrain_simplification/src/terrain_simplification.cpp

```diff
@@ -0,0 +1,258 @@
+#include <terrain_simplification/terrain_simplification.hpp>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <grid_map_cv/GridMapCvConverter.hpp>
+
+#include <ros/ros.h>
+#include <iostream>
+
+using Clock = std::chrono::high_resolution_clock;
+
+namespace terrain_simplification {
+
+TerrainSimplification::TerrainSimplification ()
+  : thread_loop_(true),
+    thread_run_(&TerrainSimplification::run, this) {
```

Comment:
I am not sure I understand why you need to run the gridmap simplification in a new thread

Reply:
As the elevation map is very large (6 m x 6 m) but slow to update, this allows me to compute and publish a submap at any (achievable) rate. The main need arises as a result of the computation being dependent on the orientation of the robot. In addition, it gives me the ability to visualize it at a higher frequency.